iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/AbstractTieLineTest.java

```diff
@@ -0,0 +1,297 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.network.tck;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.test.NoEquipmentNetworkFactory;
+import com.powsybl.iidm.network.util.SV;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * @author Florian Dupuy <florian.dupuy at rte-france.com>
+ */
+public abstract class AbstractTieLineTest {
+
+    private static final String DANGLING1_NAME = "dl1_name";
+
+    private static final String INVALID = "invalid";
+
+    private static final String TO_REMOVE = "toRemove";
+
+    private static final String DUPLICATE = "duplicate";
+
+    private Network network;
+    private VoltageLevel voltageLevelA;
+    private VoltageLevel voltageLevelB;
+
+    @BeforeEach
+    public void setUp() {
+        network = NoEquipmentNetworkFactory.create();
+        voltageLevelA = network.getVoltageLevel("vl1");
+        voltageLevelB = network.getVoltageLevel("vl2");
+    }
+
+    @Test
+    public void testTieLineAdder() {
+        double tol = 0.0000001;
+
+        double r = 10.0;
+        double r2 = 1.0;
+        double x = 20.0;
+        double x2 = 2.0;
+        double hl1g1 = 0.03;
+        double hl1g2 = 0.035;
+        double hl1b1 = 0.04;
+        double hl1b2 = 0.045;
+        double hl2g1 = 0.013;
+        double hl2g2 = 0.0135;
+        double hl2b1 = 0.014;
+        double hl2b2 = 0.0145;
+
+        // adder
+        DanglingLine dl1 = voltageLevelA.newDanglingLine()
+                .setBus("busA")
+                .setId("hl1")
+                .setEnsureIdUnicity(true)
+                .setName(DANGLING1_NAME)
+                .setR(r)
+                .setX(x)
+                .setB(hl1b1 + hl1b2)
+                .setG(hl1g1 + hl1g2)
+                .setUcteXnodeCode("ucte")
+                .add();
+        DanglingLine dl2 = voltageLevelB.newDanglingLine()
+                .setBus("busB")
+                .setId("hl2")
+                .setEnsureIdUnicity(true)
+                .setR(r2)
+                .setX(x2)
+                .setB(hl2b1 + hl2b2)
+                .setG(hl2g1 + hl2g2)
+                .add();
+
+        assertEquals(List.of(dl1, dl2), network.getDanglingLines(DanglingLineFilter.UNPAIRED));
+        assertFalse(network.getDanglingLines(DanglingLineFilter.PAIRED).iterator().hasNext());
```

Comment:
`getDanglingLines()` (without filter) is not yet tested:

```suggestion
        assertEquals(List.of(dl1, dl2), network.getDanglingLines(DanglingLineFilter.UNPAIRED));
        assertFalse(network.getDanglingLines(DanglingLineFilter.PAIRED).iterator().hasNext());
        assertEquals(List.of(dl1, dl2), network.getDanglingLines());
```